rfcs/0000-rfc-forms.md

```diff
@@ -0,0 +1,148 @@
+---
+Status: Active
+Champions: christian24
+PR: {{ update_with_pr_number }}
+---
+
+# @lit-labs/forms
+
+Add advanced form validation for web components. This is a followup to the discussion in the lit repository: https://github.com/lit/lit/discussions/2489 
+
+## Objective
+
+Provide an easy to use dev experience for validating forms with web components in them. 
+
+### Goals
+- Provide a common API, so developers can quickly start using the @lit-labs/forms package without changing their existing controls. 
+- Allow cross field validation
+- Allow async validation
+- get and set dirty, touched, valid states for individual controls
+- allow for configuration (e. g. change on `blur` or `change`)
```

Comment:
I think he refers to the model update strategy. By default the model of a control is updated as soon as the UI is updated (for an input type text, at every key stroke), but you may want to update it only on blur for example.

This is because changing the model will trigger validation, and you may not want to show errors until the user blurs a field.

In my original design this configuration was set as a parameter of the `bind` directive because that's probably a UI responsibility, but it could be set on the Reactive Controller as well (just like Angular does).

Reply:
Thanks @UserGalileo, I tried making this more specific now.